docs/cheaha/tutorial/profiling.md

```diff
@@ -0,0 +1,94 @@
+# Profiling
+
+Profiling is essential for assessing how different parts of a program execute and gathering crucial performance data. It plays a vital role in debugging, pinpointing bottlenecks, optimizing code, and scaling application performance. This analysis identifies common issues such as out-of-bound memory, segmentation faults, bus errors, and runtime overhead, enabling effective troubleshooting and improvement strategies.
+
+In research contexts, where code often performs complex or resource-intensive tasks, profiling helps identify which parts of the code consume significant compute and memory resources. This insight guides optimizations aimed at reducing execution times and enhancing the overall efficiency of the program.
+
+## Who gets benefited by Profiling?
+
+## Profiling Python Codes
+
+Three common profiling techniques used in analyzing Python codes are discussed briefly in this section.
```

Comment:
In a short paragraph, give the purpose for each. Why would I want to use time vs CPU vs memory profiling?